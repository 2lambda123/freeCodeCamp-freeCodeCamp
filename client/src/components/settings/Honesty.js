import React from 'react';
import PropTypes from 'prop-types';
import { Button, Card } from 'react-bootstrap';
import { useTranslation } from 'react-i18next';

import { FullWidthRow } from '../helpers';
import SectionHeader from './SectionHeader';
import HonestyPolicy from '../../resources/honesty-policy';

import './honesty.css';

const propTypes = {
  isHonest: PropTypes.bool,
  updateIsHonest: PropTypes.func.isRequired
};

const Honesty = ({ isHonest, updateIsHonest }) => {
  const { t } = useTranslation();
  const button = isHonest ? (
    <Button
      block={true}
      className='disabled-agreed'
      disabled={true}
      variant='primary'
    >
      <p>{t('buttons.accepted-honesty')}</p>
    </Button>
  ) : (
    <Button
      block={true}
      onClick={() => updateIsHonest({ isHonest: true })}
      variant='primary'
    >
      {t('buttons.agree')}
    </Button>
  );
  return (
    <section className='honesty-policy'>
      <SectionHeader>{t('settings.headings.honesty')}</SectionHeader>
      <FullWidthRow>
        <Card className='honesty-panel'>
          <HonestyPolicy />
        </Card>
        <br />
        {button}
      </FullWidthRow>
    </section>
  );
};

Honesty.displayName = 'Honesty';
Honesty.propTypes = propTypes;

export default Honesty;
