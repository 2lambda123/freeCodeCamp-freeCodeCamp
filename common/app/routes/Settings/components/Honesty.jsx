import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Button, Panel } from 'react-bootstrap';
import { createSelector } from 'reselect';
import { connect } from 'react-redux';

import { FullWidthRow } from '../../../helperComponents';
import { userSelector } from '../../../redux';
import academicPolicy from '../../../../resource/academicPolicy';
import { updateUserBackend } from '../../../entities/user';

const propTypes = {
  isHonest: PropTypes.bool,
  policy: PropTypes.arrayOf(PropTypes.string),
  updateUserBackend: PropTypes.func.isRequired
};

const mapStateToProps = createSelector(
  userSelector,
  ({ isHonest }) => ({
    policy: academicPolicy,
    isHonest
  })
);

const mapDispatchToProps = { updateUserBackend };

class Honesty extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      showHonesty: false
    };

    this.handleAgreeClick = this.handleAgreeClick.bind(this);
    this.handleShowClick = this.handleShowClick.bind(this);
  }

  handleAgreeClick() {
    this.props.updateUserBackend({ isHonest: true });
  }

  handleShowClick() {
    this.setState(state => ({
      ...state,
      showHonesty: !state.showHonesty
    }));
  }

  render() {
    const { policy, isHonest } = this.props;
    const isHonestAgreed = (
      <Panel bsStyle='info'>
        <p>
          You have already accepted our&nbsp;
          <a href='/academic-honesty'>
            Academic Honesty Policy
          </a>
        </p>
      </Panel>
    );
    const panelToggle = (
      <h3
        onClick={ this.handleShowClick }
        >
        View Honesty Policy
      </h3>
      );
    return (
      <div id='honesty-policy'>
        <FullWidthRow>
          <h2>Academic Honesty Policy</h2>
          {
            isHonest ?
              isHonestAgreed :
              <Panel
                collapsible={ true }
                expanded={ this.state.showHonesty }
                header={ panelToggle }
                >
                {
                  policy.map(
                    (line, i) => (
                      <p
                      dangerouslySetInnerHTML={{ __html: line }}
                      key={ '' + i + line.slice(0, 10) }
                      />
                    )
                  )
                }
                <br />
                <Button
                  block={ true }
                  bsStyle='primary'
                  onClick={ this.handleAgreeClick }
                  >
                  Agree
                </Button>
              </Panel>
          }
        </FullWidthRow>
      </div>
    );
  }
}


Honesty.displayName = 'Honesty';
Honesty.propTypes = propTypes;

export default connect(mapStateToProps, mapDispatchToProps)(Honesty);
