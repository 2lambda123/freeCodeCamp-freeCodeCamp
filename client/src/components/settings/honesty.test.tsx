import { Button } from '@freecodecamp/react-bootstrap';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import Honesty from './honesty';

describe('<Honesty />', () => {
  const utils = ShallowRenderer.createRenderer();
  const updateIsHonestMock = jest.fn();

  test('<Honesty /> snapshot when isHonest is false', () => {
    const componentToRender = (
      <Honesty isHonest={false} updateIsHonest={updateIsHonestMock} />
    );
    const view = utils.render(componentToRender);
    expect(view).toMatchSnapshot('Honesty');
  });

  test('<Honesty /> snapshot when isHonest is true', () => {
    const componentToRender = (
      <Honesty isHonest={true} updateIsHonest={updateIsHonestMock} />
    );
    const view = utils.render(componentToRender);
    expect(view).toMatchSnapshot('HonestyAccepted');
  });

  test('should call updateIsHonest method on clicking agree button', () => {
    const { root } = TestRenderer.create(
      <Honesty isHonest={false} updateIsHonest={updateIsHonestMock} />
    );

    root.findByType(Button).props.onClick();
    expect(updateIsHonestMock).toHaveBeenCalledWith({ isHonest: true });
  });
});
