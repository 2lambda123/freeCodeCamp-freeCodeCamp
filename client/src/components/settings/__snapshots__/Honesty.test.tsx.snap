// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Honesty /> <Honesty /> snapshot when isHonest is false: Honesty 1`] = `
<section
  id="honesty-policy"
>
  <SectionHeader>
    settings.headings.honesty
  </SectionHeader>
  <FullWidthRow>
    <Uncontrolled(Panel)
      className="honesty-panel"
    >
      <HonestyPolicy />
    </Uncontrolled(Panel)>
    <Spacer />
    <Button
      active={false}
      aria-pressed={false}
      block={true}
      bsClass="btn"
      bsStyle="primary"
      disabled={false}
      onClick={[Function]}
    >
      buttons.agree
    </Button>
  </FullWidthRow>
</section>
`;

exports[`<Honesty /> <Honesty /> snapshot when isHonest is true: HonestyAccepted 1`] = `
<section
  id="honesty-policy"
>
  <SectionHeader>
    settings.headings.honesty
  </SectionHeader>
  <FullWidthRow>
    <Uncontrolled(Panel)
      className="honesty-panel"
    >
      <HonestyPolicy />
    </Uncontrolled(Panel)>
    <Spacer />
    <Button
      active={false}
      aria-pressed={true}
      block={true}
      bsClass="btn"
      bsStyle="primary"
      disabled={false}
      onClick={[Function]}
    >
      buttons.accepted-honesty
    </Button>
  </FullWidthRow>
</section>
`;
